### PR TITLE
Task/apps 3320/fix user identification notification

### PR DESCRIPTION
**Pull request by anzioka**
PR:
-  Replace user identification radio buttons with regular buttons
- Use foreground service to register receivers (More reliable that using WorkManager in the background)
- Optionally display/hide dialog to prompt battery optimization exemption (Preference setting only available for >=API 23)

Screenshot_20220104-191044

Screenshot_20211216-210611

### CODE REVIEW

File: app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt [hunk 1]
```diff
@@ -20,35 +20,21 @@ class UserIdentificationActivity : AppCompatActivity() {
...
-        saveBtn = findViewById(R.id.save_btn)
-        selectUserOptions = findViewById(R.id.select_user_options)
+        childUserBtn = findViewById(R.id.child_user_btn)
+        otherUserBtn = findViewById(R.id.other_user_btn)
 
         // listeners
-        saveBtn.setOnClickListener {
-            handleOnSave()
+        otherUserBtn.setOnClickListener {
+            handleOnSave(it.id)
```
Comment (UnsungHero97) [inline app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt:28]: Is this going to automatically save when the user clicks the button and then automatically close the screen? If yes, I think the current approach is better, i.e. radio buttons along with the save button.

File: app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt [hunk 2]
```diff
@@ -20,35 +20,21 @@ class UserIdentificationActivity : AppCompatActivity() {
         setContentView(R.layout.activity_user_identification)
 
         settings = EnrollmentSettings(this)
-        saveBtn = findViewById(R.id.save_btn)
-        selectUserOptions = findViewById(R.id.select_user_options)
+        childUserBtn = findViewById(R.id.child_user_btn)
+        otherUserBtn = findViewById(R.id.other_user_btn)
```
Comment (UnsungHero97) [inline app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt:24]: What does the button look like when it is selected?

**Review by UnsungHero97 — COMMENTED** [review]
The star png is quite large at 18k; a smaller file would be better. Did you try it with SVG?

File: app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt (hunk 2 above)
Comment (anzioka) [reply app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt:24]: https://user-images.githubusercontent.com/15723547/148463797-b6157965-2fca-4ae2-9d54-c7903c5fe445.mov

File: app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt (hunk 1 above)
Comment (solomontang) [reply app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt:28]: IIRC the expectation with radio buttons is that their changes are effective immediately, without the requirement of a confirm button. Like toggling your wifi on your phone does not require a 'save' action. Usually a confirm is reserved for multiple checkbox style changes where they may have intermediate states.

**Review by solomontang — APPROVED** [review]

File: app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt (hunk 1 above)
Comment (anzioka) [reply app/src/main/java/com/openlattice/chronicle/UserIdentificationActivity.kt:28]: Yes, the changes will be automatically saved on clicking the button. Users specifically asked for that behavior.

**Review by UnsungHero97 — APPROVED** [review]

**Pull request merged**